src/xdp-app-info-flatpak.c

```diff
@@ -364,6 +370,58 @@ xdp_app_info_flatpak_validate_autostart (XdpAppInfo          *app_info,
   return TRUE;
 }
 
+static const GPtrArray *
+xdp_app_info_flaptak_get_usb_queries (XdpAppInfo *app_info)
+{
+  XdpAppInfoFlatpak *app_info_flatpak = XDP_APP_INFO_FLATPAK (app_info);
+
+  if (!app_info_flatpak->queries)
+    {
+      g_autoptr(GPtrArray) usb_queries = NULL;
+
+      usb_queries = g_ptr_array_new_with_free_func ((GDestroyNotify) xdp_usb_query_free);
+
+      g_auto(GStrv) enumerable_devices = NULL;
+      g_auto(GStrv) hidden_devices = NULL;
+
+      enumerable_devices = g_key_file_get_string_list (app_info_flatpak->flatpak_info,
+                                                       "USB Devices",
+                                                       "enumerable-devices",
+                                                       NULL, NULL);
+
+      for (size_t i = 0; enumerable_devices && enumerable_devices[i] != NULL; i++)
+        {
+          g_autoptr(XdpUsbQuery) query =
+            xdp_usb_query_from_string (XDP_USB_QUERY_TYPE_ENUMERABLE, enumerable_devices[i]);
+
+          if (query)
+            g_ptr_array_add (usb_queries, g_steal_pointer (&query));
+        }
+
+      hidden_devices = g_key_file_get_string_list (app_info_flatpak->flatpak_info,
+                                                   "USB Devices",
+                                                   "hidden-devices",
+                                                   NULL, NULL);
+
+      for (size_t i = 0; hidden_devices && hidden_devices[i] != NULL; i++)
+        {
+          g_autoptr(XdpUsbQuery) query =
+            xdp_usb_query_from_string (XDP_USB_QUERY_TYPE_HIDDEN, hidden_devices[i]);
+
+          if (query)
+            g_ptr_array_add (usb_queries, g_steal_pointer (&query));
+        }
+
+      g_message ("Found %d enumerable and %d hidden for app %s",
```

Comment:
Either make this a `g_debug` or remove it entirely?

Reply:
done

Reply:
made g_debug